sql/src/main/java/io/crate/operation/projectors/SortingTopNProjector.java

```diff
@@ -126,13 +126,6 @@ public void fail(Throwable t) {
     }
 
     @Override
-    public void repeat() {
-        final int resultSize = Math.max(pq.size() - offset, 0);
```

Comment:
As discussed: I removed the resultSize == 0 special handling so that repeat works via the RowEmitter which is always used.